src/WaterPipe/WaterPipe.php

```diff
@@ -430,6 +439,15 @@ public function run()
             $pipe[1]->_runBase($pipe[0]);
         }
     }
+    
+    /**
+     * If an exception has occured while running the pipe
+     * This method will return that exception.
+     * Otherwise, will return null.
+     */
```

Comment:
```suggestion
     * @return \Exception
     */
```

Useful for intellisense